### PR TITLE
Modpubsub 47: Create pub-sub client for module registration and message sending

### PR DESCRIPTION
**Pull request by OleksiiKuzminov**

### CODE REVIEW

File: mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java [hunk 1]
```diff
@@ -29,6 +41,128 @@
...
+      client.postPubsubPublish(eventMessage, ar -> {
+        if (ar.statusCode() == HttpStatus.HTTP_NO_CONTENT.toInt()) {
+          result.complete(true);
+        } else {
+          String message = String.format("Error during publishing Event Message in PubSub. Status code: %s . Status message: %s ", ar.statusCode(), ar.statusMessage());
+          LOGGER.error(message);
+          result.completeExceptionally(new EventSendingException(message));
+        }
+      });
+    } catch (Exception e) {
```
Comment (KaterynaSenchenko) [inline mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java]: log exception?

File: mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java [hunk 2]
```diff
@@ -29,6 +41,128 @@
...
+        result = result.thenCompose(ar -> registerSubscribers(client, descriptorHolder.getSubscriberDescriptor()));
+      }
+      return result;
+    } catch (Exception e) {
+      result.completeExceptionally(e);
+      return result;
+    }
+  }
+
+  private static CompletableFuture<Void> registerEvents(PubsubClient client, List<EventDescriptor> events) {
```
Comment (KaterynaSenchenko) [inline mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java]:
```suggestion
  private static CompletableFuture<Void> registerEventTypes(PubsubClient client, List<EventDescriptor> events) {
```

File: mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java [hunk 3]
```diff
@@ -29,6 +41,128 @@
...
+        LOGGER.info("Registering events for publishers");
+        List<EventDescriptor> eventDescriptors = descriptorHolder.getPublisherDescriptor().getEventDescriptors();
+        result = result.thenCompose(ar -> registerEvents(client, eventDescriptors))
+          .thenCompose(ar -> registerPublishers(client, descriptorHolder.getPublisherDescriptor()));
+      }
+      if (descriptorHolder.getSubscriberDescriptor() != null) {
+        result = result.thenCompose(ar -> registerSubscribers(client, descriptorHolder.getSubscriberDescriptor()));
+      }
+      return result;
+    } catch (Exception e) {
```
Comment (KaterynaSenchenko) [inline mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java]: log exception?

File: mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java [hunk 4]
```diff
@@ -29,6 +41,128 @@
...
+    }
+  }
+
+  private static CompletableFuture<Void> registerEvents(PubsubClient client, List<EventDescriptor> events) {
+    List<CompletableFuture<Boolean>> list = new ArrayList<>();
+    try {
+      for (EventDescriptor eventDescriptor : events) {
+        client.postPubsubEventTypes(null, eventDescriptor, ar -> {
+          if (ar.statusCode() == HttpStatus.HTTP_CREATED.toInt()) {
+            list.add(CompletableFuture.completedFuture(true));
```
Comment (Igor-Gorchakov) [inline mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java]: Add future to the `list` before `postPubsubEventTypes`, then pass the future to the handler to complete or fail it.

File: mod-pubsub-client/src/test/java/org/folio/util/pubsub/AbstractRestTest.java [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  protected static String okapiUrl;
+
+  @ClassRule
+  public static EmbeddedKafkaCluster cluster = provisionWith(useDefaults());
+
+  @BeforeClass
+  public static void setUpClass(final TestContext context) throws Exception {
+    vertx = Vertx.vertx();
+    runDatabase();
+    cluster.createTopic(TopicConfig.forTopic("pub-sub.diku.record_created").build());
```
Comment (Igor-Gorchakov) [inline mod-pubsub-client/src/test/java/org/folio/util/pubsub/AbstractRestTest.java]: Please move 57-60 lines into a separate method, just to have 
runDatabase();
createTopic();
deployVerticle(context);

**Pull request merged**